rocket_twin/systems/control/stage_controller_cosapp.py

```diff
@@ -0,0 +1,34 @@
+from cosapp.base import System
+
+
+class StageControllerCoSApp(System):
+    """System which allows stage command through CoSApp sequences.
+
+    Inputs
+    ------
+    is_on: int,
+        rocket's number of stages
+    weight_prop_i: float,
+        i-th stage fuel weight
+    weight_max_i: float,
+        i-th stage maximum fuel weight
+
+    Outputs
+    ------
+    w: float,
+        command flux
+    """
+
+    def setup(self):
+
+        self.add_inward("weight_prop", 0.0, desc="Stage propellant weight", unit="kg")
+        self.add_inward("weight_max", 1.0, desc="Stage maximum propellant weight", unit="kg")
+        self.add_inward("is_on", False, desc="Whether the stage is on or not")
+
+        self.add_outward("w", 1.0, desc="Command flow", unit="")
+
+        self.add_event("full", trigger="weight_prop == weight_max")
+
+    def compute(self):
+
+        self.w = self.is_on
```

Comment:
is it really needed ? what about partial thrust command (to anticipate return rocket management ...)

Reply:
Partial thrust command would be possible by adding a new inward to the controller (for instance, flux_rate) and multiplying it by is_on (so the equation would be self.w = self.is_on * self.flux_rate). I did not add it since I don't know which values this variable would take. Should I add it anyway?